SAMLPortal/Services/LdapAuthenticationService.cs

```diff
@@ -14,13 +14,24 @@ public LdapAuthenticationService()
 		{
 			_connection = new LdapConnection
 			{
-				SecureSocketLayer = false
+				SecureSocketLayer = bool.Parse(GlobalSettings.Get("LDAP_SSL"))
 			};
 		}
 
 		public AppUser Login(string username, string password)
 		{
-			_connection.Connect(GlobalSettings.Get("LDAP_Host"), GlobalSettings.GetInt("LDAP_Port"));
+			var ldapPortU = GlobalSettings.GetInt("LDAP_Port");
+			var ldapPort = 0;
+			if (ldapPortU == null)
+			{
+				throw new Exception("Error : LDAP Port not set in database. Cannot continue.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)